hazelcast/src/main/java/com/hazelcast/map/impl/query/QueryRunner.java

```diff
@@ -68,6 +75,38 @@ public QueryRunner(MapServiceContext mapServiceContext, QueryOptimizer optimizer
         this.resultProcessorRegistry = resultProcessorRegistry;
     }
 
+    /**
+     * Runs a query on a chunk of a single partition. The chunk is defined by the offset {@code tableIndex}
+     * and the soft limit {@code fetchSize}.
+     *
+     * @param query       the query
+     * @param partitionId the partition which is queried
+     * @param tableIndex  the index at which to start querying
+     * @param fetchSize   the soft limit for the number of items to be queried
+     * @return the queryied entries along with the next {@code tableIndex} to resume querying
+     */
+    public MapEntriesWithCursor runPartitionScanQueryOnPartitionChunk(Query query, int partitionId, int tableIndex,
+                                                                      int fetchSize) {
```

Comment:
Since now we've decided that the iterator doesn't support `remove`, I've now reused the `ResultProcessor` and the method returns `ResultSegment`.